File: android-citizen-app/app/src/main/java/com/sih/trial2/Volunteer.java
package com.sih.trial2;

public class Volunteer {
    private final String name;
    private final double latitude;
    private final double longitude;
    private final boolean available;
    private final String phone;
    private float distanceKm;

    public Volunteer(String name, double latitude, double longitude, boolean available, String phone) {
        this.name = name;
        this.latitude = latitude;
        this.longitude = longitude;
        this.available = available;
        this.phone = phone;
    }

    public String getName() { return name; }
    public double getLatitude() { return latitude; }
    public double getLongitude() { return longitude; }
    public boolean isAvailable() { return available; }
    public String getPhone() { return phone; }
    public float getDistanceKm() { return distanceKm; }
    public void setDistanceKm(float distanceKm) { this.distanceKm = distanceKm; }
}

File: disaster-relay/src/models/Volunteer.ts
import mongoose from "mongoose";

const volunteerSchema = new mongoose.Schema(
  {
    name: {
      type: String,
      required: true,
      trim: true,
    },

    latitude: {
      type: Number,
      required: true,
    },

    longitude: {
      type: Number,
      required: true,
    },

    available: {
      type: Boolean,
      default: true,
    },

    phone: {
      type: String,
      required: false,
    },
  },
  {
    timestamps: true,
  }
);

export default mongoose.model("Volunteer", volunteerSchema);
